### PR TITLE
(SIMP-8416) Packagecloud and rsync fixes

### PR DESCRIPTION
**Pull request by trevor-vaughan**
* Fixed:
  * rsync handling has a better check to see if rsync actually works prior to
    using it.  The old method had the potential to try and use rsync even if it
    no longer worked (FIPS flipped for example).
* Changed:
  * Migrated from PackageCloud to the SIMP download server for updates moving
    forward.

SIMP-8416 #comment simp-beaker-helpers update

### CODE REVIEW

File: lib/simp/beaker_helpers.rb [hunk 1]
```diff
@@ -584,7 +606,7 @@ def rhel_rhsm_subscribe(sut, *opts)
   end
 
   def sosreport(sut, dest='sosreports')
-    sut.install_package('sos')
+    on(sut, 'puppet resource package simp-release-community source="https://download.simp-project.com/simp-release-community.rpm" ensure=latest provider=rpm')
```
Comment (lnemsick-simp) [inline lib/simp/beaker_helpers.rb:609]: - sos is in the base CentOS package for el7.  This rpm provides yum repos for epel, postgresql, puppet, and simp (https://download.simp-project.com/simp/yum...).  Why is this needed?
- If simp-beaker-helpers was used on an isolated network, there would be no way to substitute the correct url for the simp-release-community.rpm.

File: lib/simp/beaker_helpers.rb [hunk 2]
```diff
@@ -1184,48 +1207,22 @@ def install_puppet
...
-        :sslverify => 1,
-        :sslcacert => '/etc/pki/tls/certs/ca-bundle.crt',
-        :metadata_expire => 300
-      }
-    }
-    # Verify that the repos passed to disable are in the list of valid repos
-    disable.each { |d|
-      unless repos.has_key?(d)
-        raise("ERROR: install_simp_repo - disable contains invalid SIMP repo '#{d}'.")
+  def install_simp_repos(sut, disable = [])
```
Comment (lnemsick-simp) [inline lib/simp/beaker_helpers.rb:1210]: The list of repos is different.  The documentation needs to change. Furthermore, there is no mapping of old repo keys in the disable list to the new keys in the disable list.  So, this is not backwards compatible.

**Review by lnemsick-simp — CHANGES_REQUESTED** [review]
This is not backward compatible because the list of repos that can be disabled is different.

File: lib/simp/beaker_helpers.rb (hunk 2 above)
Comment (trevor-vaughan) [reply lib/simp/beaker_helpers.rb:1210]: I'll add the mapping.

File: lib/simp/beaker_helpers.rb [hunk 3]
```diff
@@ -1170,62 +1192,57 @@ def install_puppet
...
+  #    * 'simp-community-epel'
+  #    * 'simp-community-postgres'
+  #    * 'simp-community-puppet'
+  #
+  def install_simp_repos(sut, disable = [])
+    on(sut, 'puppet resource package simp-release-community source="https://download.simp-project.com/simp-release-community.rpm" ensure=latest provider=rpm')
+
+    unless disable.empty?
+      if disable.include?('simp')
+        disable.delete('simp')
```
Comment (lnemsick-simp) [inline lib/simp/beaker_helpers.rb:1223]: should this code be modifying a passed parameter?

Comment (trevor-vaughan) [reply lib/simp/beaker_helpers.rb:1223]: @lnemsick-simp In this case it's correcting it and that change should probably be pushed back up. Would you like me to change it to a `dup`?

**Comment by trevor-vaughan** [comment]
Need to remove the calls to `yumrepo` because it doesn't exist natively in Puppet 6 :-|

**Review by lnemsick-simp — APPROVED** [review]
